### PR TITLE
Accelerate small spectra computation

### PR DESCRIPTION
**Pull request by minouHub**
### Description
Hello,

Context: I'm doing CO2 and CO fitting on small waveranges. This involves to calculate a lot of spectra for each fitting. I noticed that a CO simulation at equilibrium cost approximately 0.3-0.4 s on my computer. Approximately 0.3 s is used to retrieve the RADIS version via `get_version()`. I reckoned it would be more efficient to store the version number in the SpectrumFactory object.

**Code to test pull-request:**
```pyton
from radis import SpectrumFactory
sf = SpectrumFactory(wavenum_min=2384,
                     wavenum_max=2385,
                     molecule='CO2',
                      isotope='1,2',
                     broadening_max_width=10,  # cm-1
                     medium='vacuum',
                     verbose=3,    # more for more details
                     )
sf.fetch_databank()
s = sf.eq_spectrum(Tgas=300, path_length=1)
```
**Before:**
```python
----------------------------------------
Scaling equilibrium linestrength
0.00s - Scaled equilibrium linestrength
0.00s - Calculated lineshift
0.00s - Calculate broadening HWHM
Calculating line broadening (260 lines: expect ~ 0.03s on 1 CPU)
...... 0.00s - Precomputed DLM lineshapes (10)
...... 0.00s - Initialized vectors
...... 0.00s - Get closest matching line & fraction
...... 0.00s - Distribute lines over DLM
...... 0.00s - Convolve and sum on spectral range
0.00s - Calculated line broadening
0.00s - Calculated other spectral quantities
0.02s - Spectrum calculated (before object generation)
0.13s - Generated Spectrum object
0.15s - Spectrum calculated
```

**After:**
```python
----------------------------------------
Scaling equilibrium linestrength
0.00s - Scaled equilibrium linestrength
0.00s - Calculated lineshift
0.02s - Calculate broadening HWHM
Calculating line broadening (260 lines: expect ~ 0.03s on 1 CPU)
...... 0.01s - Precomputed DLM lineshapes (10)
...... 0.00s - Initialized vectors
...... 0.00s - Get closest matching line & fraction
...... 0.00s - Distribute lines over DLM
...... 0.00s - Convolve and sum on spectral range
0.01s - Calculated line broadening
0.00s - Calculated other spectral quantities
0.02s - Spectrum calculated (before object generation)
0.00s - Generated Spectrum object
0.02s - Spectrum calculated
```

### CODE REVIEW

**Comment by erwanp** [comment]
Wow. Never thought that this could be so slow. Good catch !

`get_version()` checks for the git commit number by default; apparently this is slow !
If we can afford to just have the RADIS version (ex : `0.9.29`, not `0.9.29-g7c1be93`), you don't need to store the version in the Factory and use `radis.__version__` instead : 

```
%timeit radis.get_version()
141 ms ± 4.77 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)

%timeit radis.get_version(add_git_number=False)
118 µs ± 1.83 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)

%timeit radis.__version__
71.3 ns ± 2.18 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)
```

**Comment by minouHub** [comment]
I ran this handy `%timeit` in debug mode in the middle of an equilibrium calculation:
```python
IPdb [1]: %timeit self.version
49.9 ns ± 2.35 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)
```
So I guess it's better to keep the full version number in the `SpectrumFactory ` and spend 0.1 ms **once** in the `SpectrumFactory ` definition

**Comment by erwanp** [comment]
Then add `version` in [radis.__init__](https://github.com/radis/radis/blob/develop/radis/__init__.py#L155) directly (`radis.__version__` won't have the git_number ; and `radis.version` will). 

It doesn't make much sense to have it at the SpectrumFactory level.

**Comment by minouHub** [comment]
Didn't know if you wanted the variable to be hidden "__complete_version__" or not. I made it public.

**Comment by minouHub** [comment]
I changed the scope of the PR to acceleration of small spectra computation. In that new context, and in the case the overpopulation feature is not used, I added a condition to avoid checking for ... overpopulation. This saved approx. 6 ms on my computer (out of 32 ms in this example). 
**Code to test pull-request:**
```python
from radis import SpectrumFactory

wmin, wmax = 2384, 2385
sf = SpectrumFactory(wavenum_min=wmin,
                     wavenum_max=wmax,
                     molecule='CO2',
                     isotope='1',
                     broadening_max_width=5,  # cm-1
                     medium='air', 
                     verbose=3,    
                     path_length=10.32, #cm
                       wstep=0.001, 
                     )
sf.fetch_databank()

Ttrans, Trot, Tvib = 2000, 2100, 2200
s_HITEMP = sf.non_eq_spectrum(
                            mole_fraction=0.2, 
                            pressure = 1,
                            Ttrans = Ttrans,
                            Trot = Trot,
                            Tvib = Tvib,
                            name='CO - Ts = ({:.0f}, {:.0f}) K'.format(Ttrans, Tvib)
                            )
```

**Before:**
```python
----------------------------------------
0.00s - Checked nonequilibrium parameters
sorting lines by vibrational bands
lines sorted in 0.0s
...... 0.26s - partition functions
...... 0.00s - populations
0.26s - Calculated nonequilibrium populations
...... 0.00s - map partition functions
...... 0.00s - corrected for populations and stimulated emission
0.00s - scaled nonequilibrium linestrength
0.00s - calculated emission integral
0.00s - Calculated lineshift
0.02s - Calculate broadening HWHM
Calculating line broadening (128 lines: expect ~ 0.06s on 1 CPU)
...... 0.00s - Precomputed DLM lineshapes (8)
...... 0.00s - Initialized vectors
...... 0.00s - Get closest matching line & fraction
...... 0.00s - Distribute lines over DLM
...... 0.02s - Convolve and sum on spectral range
...... 0.00s - Initialized vectors
...... 0.00s - Get closest matching line & fraction
...... 0.00s - Distribute lines over DLM
...... 0.02s - Convolve and sum on spectral range
0.03s - Calculated line broadening
0.00s - Calculated other spectral quantities
0.33s - Spectrum calculated (before object generation)
0.00s - Generated Spectrum object
0.33s - Spectrum calculated
```

**After:**
```python
----------------------------------------
0.00s - Checked nonequilibrium parameters
sorting lines by vibrational bands
lines sorted in 0.0s
...... 0.18s - partition functions
...... 0.00s - populations
0.18s - Calculated nonequilibrium populations
...... 0.00s - map partition functions
...... 0.00s - corrected for populations and stimulated emission
0.00s - scaled nonequilibrium linestrength
0.02s - calculated emission integral
0.00s - Calculated lineshift
0.00s - Calculate broadening HWHM
Calculating line broadening (128 lines: expect ~ 0.06s on 1 CPU)
...... 0.00s - Precomputed DLM lineshapes (8)
...... 0.00s - Initialized vectors
...... 0.00s - Get closest matching line & fraction
...... 0.00s - Distribute lines over DLM
...... 0.02s - Convolve and sum on spectral range
...... 0.00s - Initialized vectors
...... 0.00s - Get closest matching line & fraction
...... 0.00s - Distribute lines over DLM
...... 0.02s - Convolve and sum on spectral range
0.04s - Calculated line broadening
0.00s - Calculated other spectral quantities
0.27s - Spectrum calculated (before object generation)
0.00s - Generated Spectrum object
0.27s - Spectrum calculated
```

**Review by erwanp — APPROVED** [review]

**Comment by erwanp** [comment]
Well done! 

So the only thing left are the Partition Functions,  which can be greatly improved using the on-the-fly tabulation approach of @dcmvdbekerom  (proof of concept [here](https://github.com/radis/radis-benchmark/blob/master/manual_benchmarks/fast-parsum.ipynb)   (1000x faster).

Total time would become 0.09s.

File: radis/lbl/factory.py [hunk 1]
```diff
@@ -527,7 +527,7 @@ def __init__(
         self.SpecDatabase = None  # the database to store spectra. Not to be confused
         # with the databank where lines are stored
         self.database = None  # path to previous database
-
+        self.version = get_version()
```
[the remark this answers is not in the document]
Comment (minouHub) [reply radis/lbl/factory.py]: done in 1d9a3e80

**Comment by erwanp** [comment]
You can already merge this one if you want! The rest will be done in #45

**Pull request merged**